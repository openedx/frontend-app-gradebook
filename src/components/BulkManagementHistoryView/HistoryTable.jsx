/* eslint-disable react/button-has-type, import/no-named-as-default */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { DataTable } from '@edx/paragon';

import { bulkManagementColumns } from 'data/constants/app';
import selectors from 'data/selectors';

import ResultsSummary from './ResultsSummary';

export const mapHistoryRows = ({
  resultsSummary,
  originalFilename,
  user,
  ...rest
}) => ({
  resultsSummary: (<ResultsSummary {...resultsSummary} />),
  filename: (<span className="wrap-text-in-cell">{originalFilename}</span>),
  user: (<span className="wrap-text-in-cell">{user}</span>),
  ...rest,
});

/**
 * <HistoryTable />
 * Table with history of bulk management uploads, including a results summary which
 * displays total, skipped, and failed uploads
 */
export const HistoryTable = ({
  bulkManagementHistory,
}) => (
  <DataTable
    data={bulkManagementHistory.map(mapHistoryRows)}
    hasFixedColumnWidths
    columns={bulkManagementColumns}
    className="table-striped"
    itemCount={bulkManagementHistory.length}
  />
);
HistoryTable.defaultProps = {
  bulkManagementHistory: [],
};
HistoryTable.propTypes = {
  // redux
  bulkManagementHistory: PropTypes.arrayOf(PropTypes.shape({
    originalFilename: PropTypes.string.isRequired,
    user: PropTypes.string.isRequired,
    timeUploaded: PropTypes.string.isRequired,
    resultsSummary: PropTypes.shape({
      rowId: PropTypes.number.isRequired,
      text: PropTypes.string.isRequired,
    }),
  })),
};

export const mapStateToProps = (state) => ({
  bulkManagementHistory: selectors.grades.bulkManagementHistoryEntries(state),
});

export default connect(mapStateToProps)(HistoryTable);
