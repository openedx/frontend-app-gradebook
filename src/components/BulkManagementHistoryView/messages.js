import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  heading: {
    id: 'gradebook.BulkManagementHistoryView.heading',
    defaultMessage: 'Bulk Management History',
    description: 'Heading text for BulkManagement History Tab',
  },
  helpText: {
    id: 'gradebook.BulkManagementHistoryView',
    defaultMessage: 'Below is a log of previous grade imports.  To download a CSV of your gradebook and import grades for override, return to the Gradebook.  Please note, after importing grades, it may take a few seconds to process the override.',
    description: 'Bulk Management History View help text',
  },
  successDialog: {
    id: 'gradebook.BulkManagementHistoryView.successDialog',
    defaultMessage: 'CSV processing. File uploads may take several minutes to complete.',
    description: 'Success Dialog message in BulkManagement Tab File Upload Form',
  },
});

export default messages;
