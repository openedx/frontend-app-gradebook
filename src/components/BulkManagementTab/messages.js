import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  csvUploadLabel: {
    id: 'gradebook.BulkManagementTab.csvUploadLabel',
    defaultMessage: 'Upload Grade CSV',
    description: 'Button in BulkManagementTab Alerts',
  },
  heading: {
    id: 'gradebook.BulkManagementTab.heading',
    defaultMessage: 'Use this feature by downloading a CSV for bulk management, overriding grades locally, and coming back here to upload.',
    description: 'Heading text for BulkManagement Tab',
  },
  hint1: {
    id: 'gradebook.BulkManagementTab.hint1',
    defaultMessage: 'Results appear in the table below.',
    description: 'Hint text on BulkManagement Tab History Table',
  },
  hint2: {
    id: 'gradebook.BulkManagementTab.hint2',
    defaultMessage: 'Grade processing may take a few seconds.',
    description: 'Hint text on BulkManagement Tab History Table',
  },
  importBtnText: {
    id: 'gradebook.BulkManagementTab.importBtnText',
    defaultMessage: 'Import Grades',
    description: 'Button in BulkManagement Tab File Upload Form',
  },
  successDialog: {
    id: 'gradebook.BulkManagementTab.successDialog',
    defaultMessage: 'CSV processing. File uploads may take several minutes to complete.',
    description: 'Success Dialog message in BulkManagement Tab File Upload Form',
  },
});

export default messages;
