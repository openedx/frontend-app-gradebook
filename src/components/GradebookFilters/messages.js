import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  assignments: {
    id: 'gradebook.GradebookFilters.assignmentsFilterLabel',
    defaultMessage: 'Assignments',
    description: 'Assignment filter group label in Gradebook Filters',
  },
  overallGrade: {
    id: 'gradebook.GradebookFilters.overallGradeFilterLabel',
    defaultMessage: 'Overall Grade',
    description: 'Overall Grade filter group label in Gradebook Filters',
  },
  studentGroups: {
    id: 'gradebook.GradebookFilters.studentGroupsFilterLabel',
    defaultMessage: 'Student Groups',
    description: 'Student Groups filter group label in Gradebook Filters',
  },
  includeCourseTeamMembers: {
    id: 'gradebook.GradebookFilters.includeCourseTeamMembersFilterLabel',
    defaultMessage: 'Include Course Team Members',
    description: 'Include Course Team Members filter label in Gradebook Filters',
  },
  assignment: {
    id: 'gradebook.GradebookFilters.assignmentFilterLabel',
    defaultMessage: 'Assignment',
    description: 'Assignment filter select label in Gradebook Filters',
  },
  assignmentTypes: {
    id: 'gradebook.GradebookFilters.assignmentTypesLabel',
    defaultMessage: 'Assignment Types',
    description: 'Assignment Types filter select label in Gradebook Filters',
  },
  maxGrade: {
    id: 'gradebook.GradebookFilters.maxGradeFilterLabel',
    defaultMessage: 'Max Grade',
    description: 'Max-grade filter select label in Gradebook Filters',
  },
  minGrade: {
    id: 'gradebook.GradebookFilters.minGradeFilterLabel',
    defaultMessage: 'Min Grade',
    description: 'Min-grade filter select label in Gradebook Filters',
  },
  cohorts: {
    id: 'gradebook.GradebookFilters.cohorts',
    defaultMessage: 'Cohorts',
    description: 'Cohorts filter select label in Gradebook Filters',
  },
  cohortAll: {
    id: 'gradebook.GradebookFilters.cohortsAll',
    defaultMessage: 'Cohort-All',
    description: 'Cohorts filter select default in Gradebook Filters',
  },
  tracks: {
    id: 'gradebook.GradebookFilters.tracks',
    defaultMessage: 'Tracks',
    description: 'Tracks filter select label in Gradebook Filters',
  },
  trackAll: {
    id: 'gradebook.GradebookFilters.trackAll',
    defaultMessage: 'Track-All',
    description: 'Tracks filter select default in Gradebook Filters',
  },
  closeFilters: {
    id: 'gradebook.GradebookFilters.closeFilters',
    defaultMessage: 'Close Filters',
    description: 'Button label for Close button in Gradebook Filters',
  },
});

export default messages;
