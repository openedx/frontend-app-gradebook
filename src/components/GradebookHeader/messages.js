import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  backToDashboard: {
    id: 'gradebook.GradebookHeader.backButton',
    defaultMessage: 'Back to Dashboard',
    description: 'Button text to take user back to LMS dashboard in Gradebook Header',
  },
  gradebook: {
    id: 'gradebook.GradebookHeader.appLabel',
    defaultMessage: 'Gradebook',
    description: 'Top-level app title in Gradebook Header component',
  },
  frozenWarning: {
    id: 'gradebook.GradebookHeader.frozenWarning',
    defaultMessage: 'The grades for this course are now frozen. Editing of grades is no longer allowed.',
    description: 'Warning message in Gradebook Header for frozen messages',
  },
  unauthorizedWarning: {
    id: 'gradebook.GradebookHeader.unauthorizedWarning',
    defaultMessage: 'You are not authorized to view the gradebook for this course.',
    description: 'Warning message in Gradebook Header when user is not allowed to view the app',
  },
  toActivityLog: {
    id: 'gradebook.GradebookHeader.toActivityLogButton',
    defaultMessage: 'View Bulk Management History',
    description: 'Button text for button navigating to Bulk Managment Activity Log',
  },
  toGradesView: {
    id: 'gradebook.GradebookHeader.toGradesView',
    defaultMessage: 'Return to Gradebook',
    description: 'Button text for button navigating to Grades view.',
  },
});

export default messages;
