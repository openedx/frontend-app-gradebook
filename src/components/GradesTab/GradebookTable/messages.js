import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  emailHeading: {
    id: 'gradebook.GradesTab.table.headings.email',
    defaultMessage: 'Email',
    description: 'Gradebook table email column header',
  },
  totalGradeHeading: {
    id: 'gradebook.GradesTab.table.headings.totalGrade',
    defaultMessage: 'Total Grade (%)',
    description: 'Gradebook table total grade column header',
  },
  usernameHeading: {
    id: 'gradebook.GradesTab.table.headings.username',
    defaultMessage: 'Username',
    description: 'Gradebook table username column header',
  },
  studentKeyLabel: {
    id: 'gradebook.GradesTab.table.labels.studentKey',
    defaultMessage: 'Student Key*',
    description: 'Gradebook table Student Key label',
  },
  usernameLabel: {
    id: 'gradebook.GradesTab.table.labels.username',
    defaultMessage: 'Username',
    description: 'Gradebook table username label',
  },
  totalGradePercentage: {
    id: 'gradebook.GradesTab.table.totalGradePercentage',
    defaultMessage: 'Total Grade values are always displayed as a percentage',
    description: 'Gradebook table message that total grades are displayed in percent format',
  },
});

export default messages;
