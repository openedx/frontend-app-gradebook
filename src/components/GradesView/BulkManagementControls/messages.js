import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  downloadGradesBtn: {
    id: 'gradebook.GradesView.BulkManagementControls.bulkManagementLabel',
    defaultMessage: 'Download Grades',
    description: 'A labeled button that allows an admin user to download course grades all at once (in bulk).',
  },
});

export default messages;
