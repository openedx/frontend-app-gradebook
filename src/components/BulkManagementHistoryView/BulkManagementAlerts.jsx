/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from '@edx/frontend-platform/i18n';
import { connect } from 'react-redux';

import { Alert } from '@edx/paragon';

import selectors from 'data/selectors';
import messages from './messages';

/**
 * <BulkManagementAlerts />
 * Alerts to display at the top of the BulkManagement tab
 */
export const BulkManagementAlerts = ({
  bulkImportError,
  uploadSuccess,
}) => (
  <>
    <Alert
      variant="danger"
      show={!!bulkImportError}
      dismissible={false}
    >
      {bulkImportError}
    </Alert>
    <Alert
      variant="success"
      show={uploadSuccess}
      dismissible={false}
    >
      <FormattedMessage {...messages.successDialog} />
    </Alert>
  </>
);

BulkManagementAlerts.defaultProps = {
  bulkImportError: '',
  uploadSuccess: false,
};

BulkManagementAlerts.propTypes = {
  // redux
  bulkImportError: PropTypes.string,
  uploadSuccess: PropTypes.bool,
};

export const mapStateToProps = (state) => ({
  bulkImportError: selectors.grades.bulkImportError(state),
  uploadSuccess: selectors.grades.uploadSuccess(state),
});

export default connect(mapStateToProps)(BulkManagementAlerts);
