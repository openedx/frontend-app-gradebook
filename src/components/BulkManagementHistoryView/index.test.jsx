/* eslint-disable import/no-named-as-default */
import React from 'react';
import { shallow } from 'enzyme';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import { BulkManagementHistoryView } from '.';
import BulkManagementAlerts from './BulkManagementAlerts';
import HistoryTable from './HistoryTable';
import messages from './messages';

jest.mock('./BulkManagementAlerts', () => 'BulkManagementAlerts');
jest.mock('./HistoryTable', () => 'HistoryTable');

describe('BulkManagementHistoryView', () => {
  describe('component', () => {
    let el;
    beforeEach(() => {
      el = shallow(<BulkManagementHistoryView />);
    });
    describe('snapshot', () => {
      const snapshotSegments = [
        'heading from messages.BulkManagementHistoryView.heading',
        '<BulkManagementAlerts />',
        '<HistoryTable />',
      ];
      test(`snapshot - loads ${snapshotSegments.join(', ')}`, () => {
        expect(el).toMatchSnapshot();
      });
      test('heading - h4 loaded from messages', () => {
        const heading = el.find('h4');
        expect(heading.getElement()).toEqual((
          <h4>
            <FormattedMessage {...messages.heading} />
          </h4>
        ));
      });
      test('heading, then alerts, then upload form, then table', () => {
        expect(el.childAt(0).is('h4')).toEqual(true);
        expect(el.childAt(2).is(BulkManagementAlerts)).toEqual(true);
        expect(el.childAt(3).is(HistoryTable)).toEqual(true);
      });
    });
  });
});
