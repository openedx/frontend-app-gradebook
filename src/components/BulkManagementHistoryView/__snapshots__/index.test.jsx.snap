// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkManagementHistoryView component snapshot snapshot - loads heading from messages.BulkManagementHistoryView.heading, <BulkManagementAlerts />, <HistoryTable /> 1`] = `
<div
  className="bulk-management-history-view"
>
  <h4>
    <FormattedMessage
      defaultMessage="Bulk Management History"
      description="Heading text for BulkManagement History Tab"
      id="gradebook.BulkManagementHistoryView.heading"
    />
  </h4>
  <p
    className="help-text"
  >
    <FormattedMessage
      defaultMessage="Below is a log of previous grade imports.  To download a CSV of your gradebook and import grades for override, return to the Gradebook.  Please note, after importing grades, it may take a few seconds to process the override."
      description="Bulk Management History View help text"
      id="gradebook.BulkManagementHistoryView"
    />
  </p>
  <BulkManagementAlerts />
  <HistoryTable />
</div>
`;
